client/engine/chainservice/eth_chainservice.go

```diff
@@ -219,33 +219,35 @@ func (ecs *EthChainService) subscribeForLogs() {
 	if err != nil {
 		ecs.fatalF("subscribeFilterLogs failed: %w", err)
 	}
-	for {
-		select {
-		case <-ecs.ctx.Done():
-			sub.Unsubscribe()
-			return
-		case err := <-sub.Err():
-			if err != nil {
-				ecs.fatalF("received error from the subscription channel: %w", err)
-			}
+	go func() {
```

Comment:
Nit but it might be nice to add a comment explaining that we're waiting for the synchronous subscribe before listening async

Reply:
Added in 7a274372.